src/shaders/atmosphere.fragment.glsl

```diff
@@ -0,0 +1,136 @@
+uniform float u_transition;
+uniform highp float u_fadeout_range;
+uniform highp float u_temporal_offset;
+uniform vec3 u_start_color;
+uniform vec4 u_color;
+uniform vec4 u_space_color;
+uniform vec4 u_high_color;
+uniform float u_star_intensity;
+uniform float u_star_size;
+uniform float u_star_density;
+uniform float u_horizon_angle;
+uniform mat4 u_rotation_matrix;
+
+varying highp vec3 v_ray_dir;
+varying highp vec3 v_horizon_dir;
+
+float random(vec3 p) {
+    p = fract(p * vec3(23.2342, 97.1231, 91.2342));
+    p += dot(p.zxy, p.yxz + 123.1234);
+    return fract(p.x * p.y);
+}
+
+float stars(vec3 p, float scale, vec2 offset) {
+    vec2 uv_scale = (u_viewport / u_star_size) * scale;
+    vec3 position = vec3(p.xy * uv_scale + offset * u_viewport, p.z);
+
+    vec3 q = fract(position) - 0.5;
+    vec3 id = floor(position);
+
+    float random_visibility = step(random(id), u_star_density);
+    float circle = smoothstep(0.5 + u_star_intensity, 0.5, length(q));
+
+    return circle * random_visibility;
+}
```

Comment:
This produces a really nice looking starfield! Artefacts from distributing points on the cube (instead of uniform distribution on a sphere) are pretty much indistinguishable!